spring-cloud-starter-stream-processor-tasklaunchrequest-transform/src/main/java/org/springframework/cloud/stream/app/tasklaunchrequest/transform/processor/TasklaunchrequestTransformProcessorConfiguration.java

```diff
@@ -93,7 +99,10 @@ public Object setupRequest(String message) {
 				deploymentProperties,
 				applicationName);
 
-		return request;
+		Message messageResponse = MessageBuilder.withPayload(request).
+				copyHeaders(message.getHeaders()).
+				removeHeader(MessageHeaders.CONTENT_TYPE).build();
```

Comment:
For the record, my question about why this needs to happen still stands.

Reply:
@olegz ,

That helps, thanks. 

Looks like this is an answer as well, @mminella : https://docs.spring.io/spring-cloud-stream/docs/Elmhurst.M3/reference/htmlsingle/#mime-types-and-java-types

But it is still a part of the lost discussion and there is no any explanation in the commit message. We definitely will lose this PR eventually and that will be fully unclear from commit history "why?". Therefore my concern about honoring Open Source principles.

From here I see we bump with the bigger problem: each `processor` application is going to suffer the same request `contentType` header coercion issue.

Shouldn't we come up with some general solution to allow to exclude undesired headers via configuration properties, for example?

Thanks

Reply:
I am not sure this is all that counter intuitive.
In a transformer that is encapsulated inside of a microservice i would be vert surprised if input content type was automatically included as output content type. It's a "transformer", so the expectation should be for the end user to provide one of rely on default which in our case is `a/json`

Reply:
I have to second the comment @artembilan made about why this code is needed.  While I get this from the perspective of order of precedence, etc.  However, as a user of Spring Cloud Stream, I wouldn't expect to need this code in *all* of my transformers in order to be safe (which it needs to be since you have the risk of having the content type clobbered by an upstream value).

To be clear, I don't even understand why this transformer needs to work with `Message` at all.  We return a POJO and we don't care about the input request.  I'd expect this to be handled automagically by Spring Integration or Spring Cloud Stream.

Reply:
Ok, just had a chat with Michael, so here is the verdict. 
If a transformer operating at the Message level (receives Message -> outputs Message) then the above code is correct. However. . . 
Does the aforementioned transformer have to operate at the Message level? The answer is NO.
So , rephrasing the first sentence. . .
If a transformer operating at the POJO level (receives POJO -> outputs POJO) then the above code is out of scope and the problem is pushed to SCSt. And in SCSt (after some recent improvements) we will not propagate inbound content-type header to the outbound. So the resulting message will go out with no content-type set. If it is not the case, raise the issue in SCSt

Reply:
Sounds good. So, let's go ahead then!

And I see what you mean about not propagated headers. It is about `SpringIntegrationProperties` and its:
```
/**
 * Message header names that will NOT be copied from the inbound message.
 */
private String[] messageHandlerNotPropagatedHeaders = new String[] { MessageHeaders.CONTENT_TYPE };
```

Then we definitely should step back and figure out why @cppwfs has been forced to do all this message building gymnastics...